src/main/java/org/teleight/teleightbots/bot/manager/BotManagerImpl.java

```diff
@@ -33,14 +33,27 @@ public void registerLongPolling(@NotNull String token, @NotNull String username,
         final TelegramBot bot = botProvider.provide(token, username, updateProcessor, botSettings);
 
         updateProcessor.setBot(bot);
-        bot.connect();
+        updateProcessor.start().whenComplete((user, throwable) -> {
+            if (throwable != null) {
+                System.out.println("Error while starting the bot: " + bot.getBotUsername());
+                if (botSettings.silentlyThrowMethodExecution()) {
+                    TeleightBots.getExceptionManager().handleException(throwable);
+                }
+                bot.shutdown();
+                return;
+            }
```

Comment:
Not sure if handling the exception inside `BotManagerImpl` is better design-wise instead of handling it inside the `LongPollingUpdateProcessor` itself. Also, the `user` variable isn't really used